File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitAssetWorkSheet/EngineerVisitAssetWorkSheetValidatorComCat.vb
Namespace Entity

Namespace EngineerVisitAssetWorkSheets

        Public Class EngineerVisitAssetWorkSheetValidatorComCat
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerVisitAssetWorkSheet As EngineerVisitAssetWorkSheet, ByVal ApplianceTested As String)

                'make sure that contact object is valid
                If oEngineerVisitAssetWorkSheet.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerVisitAssetWorkSheet.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If


                If ApplianceTested <> "No" Then

                    If oEngineerVisitAssetWorkSheet.AssetID = Nothing Then
                        Me.AddCriticalMessage("Appliance Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.LandlordsApplianceID = Nothing Then
                        Me.AddCriticalMessage("Landlords Appliance  Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.ApplianceTestedID = Nothing Then
                        Me.AddCriticalMessage("Appliance tested Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.FlueFlowTestID = Nothing Then
                        Me.AddCriticalMessage("Gas hose field Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.SpillageTestID = Nothing Then
                        Me.AddCriticalMessage("Electrical isolator field Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID = Nothing Then
                        Me.AddCriticalMessage("Manufactures info field Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID = Nothing Then
                        Me.AddCriticalMessage("Gas isolation Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID = Nothing Then
                        Me.AddCriticalMessage("FSP fitted Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID = Nothing Then
                        Me.AddCriticalMessage("pipework gastight Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID = Nothing Then
                        Me.AddCriticalMessage("Appliance service Missing")
                    End If

                    If oEngineerVisitAssetWorkSheet.ApplianceSafeID = Nothing Then
                        Me.AddCriticalMessage("Appliance safety Missing")
                    End If

                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitAssetWorkSheet/EngineerVisitAssetWorkSheetValidatorComCat.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitAssetWorkSheets
    {
        // make sure that contact object is valid
        public class EngineerVisitAssetWorkSheetValidatorComCat : BaseValidator
        {
            public void Validate(EngineerVisitAssetWorkSheet oEngineerVisitAssetWorkSheet, string ApplianceTested)
            {
                // make sure that contact object is valid
                if (oEngineerVisitAssetWorkSheet.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oEngineerVisitAssetWorkSheet.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if ((ApplianceTested ?? "") != "No")
                {
                    if (oEngineerVisitAssetWorkSheet.AssetID == default)
                    {
                        AddCriticalMessage("Appliance Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.LandlordsApplianceID == default)
                    {
                        AddCriticalMessage("Landlords Appliance  Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.ApplianceTestedID == default)
                    {
                        AddCriticalMessage("Appliance tested Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.FlueFlowTestID == default)
                    {
                        AddCriticalMessage("Gas hose field Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.SpillageTestID == default)
                    {
                        AddCriticalMessage("Electrical isolator field Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID == default)
                    {
                        AddCriticalMessage("Manufactures info field Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID == default)
                    {
                        AddCriticalMessage("Gas isolation Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID == default)
                    {
                        AddCriticalMessage("FSP fitted Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID == default)
                    {
                        AddCriticalMessage("pipework gastight Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID == default)
                    {
                        AddCriticalMessage("Appliance service Missing");
                    }

                    if (oEngineerVisitAssetWorkSheet.ApplianceSafeID == default)
                    {
                        AddCriticalMessage("Appliance safety Missing");
                    }
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}